packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,184 @@
+      }
+    }
+  };
+
+  handleClear = () => {
+    const { onChange, onSubmit } = this.props;
+    const searchFunction = onSubmit || onChange;
+
+    this.setState({
+      searchTerm: ''
+    });
+
+    // 1) Cancel debounced search
+    // 2) Use onClear instead of onSubmit | onChange function
+    searchFunction('');
+
+    if (this.inputRef.current) {
+      this.inputRef.current.focus();
+    }
+  };
+
+  handleCloseIconKeyDown = key => {
+    if (key.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    this.setState(
+      prevState => ({
+        isInCompactMode: !prevState.isInCompactMode
+      }),
+      () => {
+        if (!this.state.isInCompactMode && this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }
+    );
+  };
+
+  render() {
+    const { className, placeholder, value, loading, size, error } = this.props;
+    const { searchTerm, isInCompactMode } = this.state;
+
+    const isCompactSize = size === 'compact';
+    const shouldDisplayCloseButton = searchTerm && !loading && !isInCompactMode;
+
+    return (
+      <span className={className}>
+        <div className={styles[`${baseClass}__container`]}>
+          <SearchIcon
+            fill="#424d57"
+            width="18px"
+            height="18px"
+            className={cx(`lc-${baseClass}__icon--search`, {
+              [`lc-${baseClass}__icon--search-compact`]: isCompactSize
+            })}
+            onClick={isCompactSize ? this.toggleCompactMode : null}
+          />
+          {loading && (
+            <Loader
+              size="small"
+              className={styles[`${baseClass}__icon--loader`]}
+            />
+          )}
+          <Input
+            placeholder={placeholder}
+            ref={this.inputRef}
+            value={value || this.state.searchTerm}
+            onChange={this.handleChange}
+            onKeyDown={this.handleKeyPress}
+            className={cx(`lc-${baseClass}__input`, {
+              [`lc-${baseClass}__input-compact`]: isInCompactMode
+            })}
+          />
+          {shouldDisplayCloseButton && (
+            <CloseIcon
+              fill="#424d57"
+              width="18px"
+              height="18px"
+              onClick={this.handleClear}
+              onKeyDown={this.handleCloseIconKeyDown}
+              className={styles[`${baseClass}__icon--close`]}
+              tabIndex="0"
+            />
+          )}
+        </div>
+        {error && <FieldError>{error}</FieldError>}
+      </span>
+    );
+  }
+}
+
+const basePropTypes = {
+  className: PropTypes.string,
+  placeholder: PropTypes.string,
+  /**
+   * Pass this prop if you want to control searchbar term from parent component
+   */
+  value: PropTypes.string,
+  loading: PropTypes.bool,
+  /**
+   * Pass size of `compact` if you want to display extendable searchbar (usually for smaller screens)
+   */
+  size: PropTypes.oneOf(acceptedSizes),
```

Comment:
rather than using `size` property for this collapsed input I would use `collapsable: boolean` 
I see on design smaller input (for tables etc.). The size of text and high of input should depend on this `size` property instead of collapse mode.

Reply:
SGTM ☝️

Reply:
But but, it was like that earlier and you wanted a `size` property :D I'll change it back then